components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -1660,8 +1654,44 @@ func TestCreatePolicy(t *testing.T) {
 
 			assertNoPolicyChange(t, store, func() {
 				resp, err := store.CreatePolicy(ctx, &pol)
-				assert.Error(t, err)
+				require.Error(t, err)
 				assert.Nil(t, resp)
+				_, ok := err.(*storage_errors.ErrForeignKey)
+				require.True(t, ok, "expected foreign key error")
+				assert.Equal(t, "project not found: not-real-project", err.Error())
+			})
+
+			assertEmpty(t,
+				db.QueryRow(`SELECT count(*) FROM iam_policies WHERE id=$1 AND name=$2 AND type=$3`,
+					polID, name, typeVal.String()))
+			assertEmpty(t, db.QueryRow(`SELECT count(*) FROM iam_statements WHERE policy_id=policy_db_id($1)`, polID))
+			assertEmpty(t, db.QueryRow(`SELECT count(*) FROM iam_policy_members WHERE policy_id=policy_db_id($1)`, polID))
+			assertEmpty(t, db.QueryRow(`SELECT count(*) FROM iam_members`))
+			assertEmpty(t, db.QueryRow(`SELECT count(*) FROM iam_policy_projects WHERE policy_id=policy_db_id($1)`, polID))
+		},
+		"policy with one existent, one non-existent project fails": func(t *testing.T) {
+			polID := genSimpleID(t, prngSeed)
+
+			projID0 := "special-project"
+			insertTestProject(t, db, projID0, "too special", storage.Custom)
+			projID1 := "not-real-project"
+
+			name, members, typeVal := "toBeCreated", []storage.Member{}, storage.Custom
+			pol := storage.Policy{
+				ID:       polID,
+				Name:     name,
+				Type:     typeVal,
+				Members:  members,
+				Projects: []string{projID0, projID1},
+			}
+
+			assertNoPolicyChange(t, store, func() {
+				resp, err := store.CreatePolicy(ctx, &pol)
+				require.Error(t, err)
+				assert.Nil(t, resp)
+				_, ok := err.(*storage_errors.ErrForeignKey)
+				require.True(t, ok, "expected foreign key error")
+				assert.Equal(t, "project not found: not-real-project", err.Error())
```

Comment:
This is new -- before, we would have only said "your projects contain one that doesn't exist". Now, we're explicit.